src/main/kotlin/calculator/UnparsedExpression.kt

```diff
@@ -0,0 +1,43 @@
+package calculator
+
+class UnparsedExpression(
+    var text: String? = null,
+) {
+    init {
+        if (text.isNullOrEmpty() || text.isNullOrBlank()) {
+            text = "0"
+        }
+    }
+
+    fun splitText(): TextTokens {
+        val value = requireNotNull(text) { "입력값을 확인하세요" }
+        val textTokens = splitByCustomDelimiter()
+        if (textTokens.isNotEmpty()) {
+            return addTextTokens(textTokens)
+        }
+        val tokens = value.split(DEFAULT_DELIMITER.toRegex())
+        return addTextTokens(tokens)
+    }
+
+    private fun addTextTokens(textTokens: List<String>): TextTokens {
+        val tokens = TextTokens()
+        for (text in textTokens) {
+            tokens.addToken(text)
+        }
+        return tokens
+    }
+
+    private fun splitByCustomDelimiter(): List<String> {
+        val result = text?.let { Regex(CUSTOM_DELIMITER).find(it) }
+        result?.let {
+            val customDelimiter = it.groupValues[1]
+            return it.groupValues[2].split(customDelimiter)
+        }
+        return emptyList()
```

Comment:
elvis(?:) 연산을 활용하면 아래와 같이 작성이 가능해보입니다!

```suggestion
        return result?.let {
            val customDelimiter = it.groupValues[1]
            it.groupValues[2].split(customDelimiter)
        } ?: emptyList()
```